File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/project/events/ProjectEvent.scala
/**
 * This file is part of mycollab-web.
 *
 * mycollab-web is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-web is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-web.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.module.project.events

import com.esofthead.mycollab.eventmanager.ApplicationEvent
import com.esofthead.mycollab.module.project.domain.ProjectGenericItem

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */
object ProjectEvent {

  class GotoAdd(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoMyProject(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoTagListView(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoFavoriteView(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class SelectFavoriteItem(source: AnyRef, data: ProjectGenericItem) extends ApplicationEvent(source, data) {}

  class TimeLoggingChangedEvent(source: AnyRef) extends ApplicationEvent(source, null) {}

  class GotoCalendarView(source: AnyRef) extends ApplicationEvent(source, null) {}

  class GotoGanttChart(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoDashboard(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoUserDashboard(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

}

File: mycollab-web/src/main/java/com/mycollab/module/project/event/ProjectEvent.kt
/**
 * mycollab-web - Parent pom providing dependency and plugin management for applications
		built with Maven
 * Copyright © ${project.inceptionYear} MyCollab (support@mycollab.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.module.project.event

import com.mycollab.vaadin.event.ApplicationEvent
import com.mycollab.module.project.domain.ProjectGenericItem

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object ProjectEvent {
    class GotoAdd(source: Any, val data: Any?) : ApplicationEvent(source)

    class GotoEdit(source: Any, val data: Any?) : ApplicationEvent(source)

    class GotoList(source: Any, val data: Any?) : ApplicationEvent(source)

    class GotoMyProject(source: Any, val data: Any?) : ApplicationEvent(source)

    class GotoTagListView(source: Any, val data: Any?) : ApplicationEvent(source)

    class GotoFavoriteView(source: Any, val data: Any?) : ApplicationEvent(source)

    class SelectFavoriteItem(source: Any, val data: ProjectGenericItem) : ApplicationEvent(source)

    class TimeLoggingChangedEvent(source: Any) : ApplicationEvent(source)

    class GotoCalendarView(source: Any) : ApplicationEvent(source)

    class GotoGanttChart(source: Any, val data: Any?) : ApplicationEvent(source)

    class GotoDashboard(source: Any, val data: Any?) : ApplicationEvent(source)

    class GotoUserDashboard(source: Any, val data: Any?) : ApplicationEvent(source)
}